java-manual/modules/ROOT/pages/query-simple.adoc

```diff
@@ -0,0 +1,363 @@
+
+var result = driver.executableQuery("""
+    MATCH (p:Person {name: $name})
+    MERGE (p)-[:KNOWS]->(:Person {name: $friend})
+    """)
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .withParameters(Map.of("name", "Alice", "friend", "Bob"))
+    .execute();
+var summary = result.summary();
+System.out.println("Query updated the database?");
+System.out.println(summary.counters().containsUpdates());
+----
+
+[WARNING]
+====
+It might feel tempting to create new relationships with a single `MERGE` clause, such as: +
+`MERGE (:Person {name: "Alice"})-[:KNOWS]->(:Person {name: "Bob"})`. +
+However, this would result in the creation of an _extra_ `Alice` node, so that you would end up with unintended duplicate records.
+To avoid this, always `MATCH` the elements that you want to update, and use the resulting reference in the `MERGE` clause (as shown in the previous example).
+See link:https://neo4j.com/developer/kb/understanding-how-merge-works/[Understanding how MERGE works].
+====
+
+
+== Delete from the database
+To remove a node and any relationship attached to it, use the Cypher clause link:{neo4j-docs-base-uri}/cypher-manual/current/clauses/delete/[`DETACH DELETE`]:
+
+.Remove the `Alice` node
+[source, java]
+----
+// import org.neo4j.driver.QueryConfig;
+
+// This does not delete _only_ p, but also all its relationships!
+var result = driver.executableQuery("""
+    MATCH (p:Person {name: $name})
+    DETACH DELETE p
+    """)
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .withParameters(Map.of("name", "Alice"))
+    .execute();
+var summary = result.summary();
+System.out.println("Query updated the database?");
+System.out.println(summary.counters().containsUpdates());
+----
+
+
+[#query-parameters]
+== Query parameters
+
+*Do not hardcode or concatenate parameters directly into queries*.
+Instead, always use placeholders and specify the link:{neo4j-docs-base-uri}/cypher-manual/current/syntax/parameters/[Cypher parameters], as shown in the previous examples.
+This is for:
+
+1. *performance benefits*: Neo4j compiles and caches queries, but can only do so if the query structure is unchanged;
+2. *security reasons*: see link:https://neo4j.com/developer/kb/protecting-against-cypher-injection/[protecting against Cypher injection].
+
+You may provide query parameters as a map through the `.withParameters()` method.
+
+[source, java]
+----
+var result = driver.executableQuery("MATCH (p:Person {name: $name})")
+    .withParameters(Map.of("name", "Alice"))  // mark-line
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute();
+----
+
+[NOTE]
+There can be circumstances where your query structure prevents the usage of parameters in all its parts.
+For those advanced use cases, see xref:query-advanced#_dynamic_values_in_property_keys_relationship_types_and_labels[Dynamic values in property keys, relationship types, and labels].
+
+
+== Error handling
```

Comment:
This feels very odd to me. This is the case for all methods in all programs. 

I would rewrite this something along the following idea:


Running a query against a remote server can fail for a number of reasons. Some of them are due to syntactically wrong queries and similar, client side issues, some of them can be caused by network issues. The driver will retry running your query in many scenarios, such as temporary server unavailability. 
For all client side exceptions it is up to you if program defensive (i.e. check if records are available to avoid `[NoSuchRecordException](https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/exceptions/NoSuchRecordException.html)`), or if you just catch and handle them).

Reply:
I agree with you that I find this section border-line confusing. The reason it is here is user interviews from a few years back, where people were lamenting not knowing how to do "proper error handling", which I interpreted as "should I differentiate between the different exceptions the driver may raise, or how should I do?". But maybe I should have just ignored that part of user confusion since, as you say, this is how error handling in _any_ software works. We'd show it in the full examples anyway.